src/test/java/org/example/controllers/ReaderControllerTest.java

```diff
@@ -0,0 +1,174 @@
+package org.example.controllers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Optional;
+import org.example.configuration.DateTimeFormatterConfiguration;
+import org.example.dto.NewReaderDTO;
+import org.example.entity.Reader;
+import org.example.services.ReaderService;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+
+import static org.example.util.Util.getErrorResponseFromMvcResult;
+import static org.example.util.Util.getFistReader;
+import static org.example.util.Util.getResponseForIdZeroOrLess;
+import static org.example.util.Util.getResponseForInvalidDecimalId;
+import static org.example.util.Util.getResponseForInvalidFieldsNewReaderDTO;
+import static org.example.util.Util.getTestReaders;
+import static org.example.util.Util.setIdForTestReaders;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+
+@AutoConfigureMockMvc
+@ExtendWith(SpringExtension.class)
+@WebMvcTest(ReaderController.class)
+@Import(DateTimeFormatterConfiguration.class)
+class ReaderControllerTest {
+
+  private static final String REQUEST_PATH = "/api/v1/readers";
+
+  @Autowired
+  MockMvc mvc;
+  @Autowired
+  ObjectMapper objectMapper;
+  @MockBean
+  ReaderService readerService;
+  @Autowired
+  DateTimeFormatter dateTimeFormatter;
+
+  @Test
+  void shouldReturnNotFoundWhenReaderNotPresent() throws Exception {
+    var idTen = 10L;
```

Comment:
I would name the variable something like `var notExistingId = 10L;` - just to make the test flow even more readable and clear.